### DISCUSSION TITLE
Ability to jump to a turn via UI when viewing a game

### DISCUSSION THREAD

**Discussion by mawilson** [BattlesnakeOfficial/feedback#171] [Feature Requests]
Some games run pretty long, & it takes a while for them to run through to the end, even when holding the arrow key down to speed through turns. It could be nice to have a button that skips playback ahead to the final turn of the game.

More robustly, a bar similar to one as with video playback, where the user could simply click along the bar & be taken to the corresponding turn. Mockup
![snakeGameVideoBar](https://user-images.githubusercontent.com/2830497/153503229-16f300d7-9195-4a4b-a8c4-c8f4da50df10.png)

The former is obviously easier than the latter - even just an ability to jump to the end could be nice.

**Comment by bvanvugt** [comment]
Huh 🤔 We do actually support this, but we don't expose it anywhere in the UI.

You can do `turn=32` on any game URL to load a specific frame.
ie: https://play.battlesnake.com/g/9d0f1db9-371d-4a73-9bbf-a60440fb36b6/?turn=32

I love the idea of a scrubbable timeline though... that'd be awesome.

**Comment by mawilson** [comment]
Yeah, the turn query string is useful. Is there anywhere in the game UI that the last turn is displayed? In the absence of knowing, I tried inputting an arbitrarily large turn (later than the last turn in the game), & that took me back to the first turn. It might be more useful in that case to go to the last turn instead.

**Comment by BoldBigflank** [comment]
There are a lot of design decisions made to prevent "spoiling" the results of the game. The length of the game, the health of snakes past the latest turn, etc, are being set up in a way to not let you know what is going to happen in that game. That works out well for a shoutcast type use case, but when people are debugging their snakes they would benefit from the whole picture at once.

**Reply by bvanvugt** [reply]
An excellent point! The casters don't even use this view anymore, so maybe we could abandon some of these design decisions in favor of DX 👍

**Comment by coreyja** [comment]
Love this idea! Hacked on it in the Board repo: https://github.com/BattlesnakeOfficial/board/pull/49

Here's a demo!
![2022-02-11 20 18 09](https://user-images.githubusercontent.com/964771/153690737-004c2385-ac5b-4a7d-b7e4-304e78fe9908.gif)

**Comment by bvanvugt** [comment]
**We've deployed Corey's version of this as an experimental feature as of 1.7.46!** 

It can be enabled by default in playback settings and persisted with your other playback preferences. Let us know what you think!

![game-scrubber-experiment](https://user-images.githubusercontent.com/1531419/153959110-0dd40128-590e-466f-b186-15e2a0eb3e06.gif)

**Reply by bvanvugt** [reply]
**Bug:** there's some funkiness in the analysis graphs when scrubbing forward/backward.

![scrubbed-graphs](https://user-images.githubusercontent.com/1531419/153960127-5691589f-57f1-4c64-976f-dec7cc94f919.gif)

**Reply by coreyja** [reply]
Oh wow that's fun 
You can build your own graph! 🤣

**Reply by coreyja** [reply]
When it rains it pours 😆 , I just noticed some related other weird behavior with these graphs

In Discord a link to a game with the `?turn` query param was posted: https://play.battlesnake.com/g/42dba90e-53d5-459f-a1e6-180315026312/?turn=801

If you open this game up and use the forwards buttons, then back buttons to go back FARTHER than the turn the query param brought you it starts moving forward again


![2022-02-14 18 26 09](https://user-images.githubusercontent.com/964771/153964065-ce52fc5e-48b0-46b5-89a7-13fed62f79e9.gif)
